File: packages/expo-modules-core/android/src/main/java/expo/modules/core/Promise.java
package expo.modules.core;

import expo.modules.core.interfaces.CodedThrowable;
import kotlin.Deprecated;

@Deprecated(message = "AsyncFunction will crash when called. Use expo.modules.kotlin.Promise instead")
public interface Promise {
  String UNKNOWN_ERROR = "E_UNKNOWN_ERROR";

  void resolve(Object value);

  void reject(String code, String message, Throwable e);

  // Obsolete methods, however nice-to-have when porting React Native modules to Expo modules.
  default void reject(Throwable e) {
    if (e instanceof CodedThrowable) {
      CodedThrowable codedThrowable = (CodedThrowable) e;
      reject(codedThrowable.getCode(), codedThrowable.getMessage(), e);
    } else {
      reject(UNKNOWN_ERROR, e);
    }
  }

  default void reject(String code, String message) {
    reject(code, message, null);
  }

  default void reject(String code, Throwable e) {
    reject(code, e.getMessage(), e);
  }
}

File: android/versioned-abis/expoview-abi46_0_0/src/main/java/abi46_0_0/expo/modules/kotlin/Promise.kt
package abi46_0_0.expo.modules.kotlin

import abi46_0_0.expo.modules.kotlin.exception.CodedException

interface Promise {
  fun resolve(value: Any?)

  fun reject(code: String, message: String?, cause: Throwable?)

  fun reject(exception: CodedException) {
    reject(exception.code, exception.localizedMessage, exception.cause)
  }
}
